src/main/java/org/embulk/parser/jsonpath/JsonpathParserPlugin.java

```diff
@@ -172,37 +172,19 @@ public void run(TaskSource taskSource, Schema schema,
                     continue;
                 }
 
-                if (!json.isArray()) {
-                    skipOrThrow(new JsonRecordValidateException(format(Locale.ENGLISH,
-                            "Json string is not representing array value json='%s'", json)), stopOnInvalidRecord);
-                    continue;
-                }
-
-                for (JsonNode recordValue : json) {
-                    try {
-                        if (recordValue.getNodeType() != JsonNodeType.OBJECT) {
-                            throw new JsonRecordValidateException(format(Locale.ENGLISH,
-                                    "Json string is not representing map value json='%s'", recordValue));
+                if (json.isArray()) {
+                    for (JsonNode recordValue : json) {
+                        try {
+                            createRecordFromJson(recordValue, schema, jsonPathMap, visitor, pageBuilder);
                         }
-
-                        for (Column column : schema.getColumns()) {
-                            JsonNode value = null;
-                            if (jsonPathMap.containsKey(column)) {
-                                try {
-                                    value = JsonPath.using(JSON_PATH_CONFIG).parse(recordValue).read(jsonPathMap.get(column));
-                                }
-                                catch (PathNotFoundException e) {
-                                    // pass (value is nullable)
-                                }
-                            }
-                            else {
-                                value = recordValue.get(column.getName());
-                            }
-                            visitor.setValue(value);
-                            column.visit(visitor);
+                        catch (DataException e) {
+                            skipOrThrow(e, stopOnInvalidRecord);
+                            continue;
                         }
-
-                        pageBuilder.addRecord();
+                    }
+                } else {
```

Comment:
I think this part needs to check whether `json` is an Object or not, like the below.
Does the current code handle properly the following input?

```json
[ { "name": "bob", "age":24 },{ "name":"alice", "age":25 } ]
```
* jsonpath: `$[1].name`  # bob

```java
} else ( json.isObject() ) { // please fix method name if this method incorrect.
    //
} else {
    // raise Exception.
}
```

Reply:
Does `jsonpath: $[1].name` mean `root: $[1].name` ?
If json is neither an object nor an array, an error will occur here.
https://github.com/hiroyuki-sato/embulk-parser-jsonpath/blob/006de80b4035393f7502cc8462b79e26280f7984/src/main/java/org/embulk/parser/jsonpath/JsonpathParserPlugin.java#L215-L219

Please see also this test.
https://github.com/hiroyuki-sato/embulk-parser-jsonpath/blob/006de80b4035393f7502cc8462b79e26280f7984/src/test/java/org/embulk/parser/jsonpath/TestJsonpathParserPlugin.java#L475-L489